File: week1/java/week1_16.java
import java.util.Queue;
import java.util.LinkedList;
class Solution {
    boolean solution(String s) {
        Queue<Character> q = new LinkedList<>();
        for(int i=0; i<s.length(); i++){
            char cur = s.charAt(i);
            if(cur == '('){
                q.add(cur);
            }else{
                if(q.size()!=0) ((LinkedList<Character>) q).removeLast();
                else return false;
            }
        }

        if(q.size() == 0) return true;
        else return false;
    }
}

File: week1/c++/week1_16.cpp
#include <string>

using namespace std;

bool solution(string s)
{
    int a = 0;
    int b = 0;
    bool answer = true;

    for (int i = 0; i < s.length(); i++)
    {
        if (a < b)
        {
            answer = false;
            break;
        }

        if (s[i] == '(')
            a++;
        else
            b++;
    }

    if (a != b)
    {
        answer = false;
    }

    return answer;
}